mlir/lib/Dialect/Rock/IR/RockDialect.cpp

```diff
@@ -1444,6 +1450,23 @@ LogicalResult ThreadwiseReadIntoOp::verify() {
   } else if (inputShape.size() != extraIdxCount + 1) {
     return emitOpError("source view must be extraIndices + 1");
   }
+
+  // Add more constraints if we see vector buffers (e.g.,
+  // memref<Kxvector<vxf16>>)
+  VectorType srcVectorType = srcType.getElementType().dyn_cast<VectorType>();
+  VectorType dstVectorType = destType.getElementType().dyn_cast<VectorType>();
+  if ((srcVectorType || dstVectorType) &&
+      gpuSrcMemSpaceAttr.getValue() != gpu::AddressSpace::Workgroup)
+    return emitOpError("Vector buffers are only allowed when we read from LDS");
```

Comment:
Just curious... why ?

Reply:
The global buffer is usually read/written from plain `memref<shapextype>` types , while when we read from LDS, the buffers can be either a plain memref (for Kpack==1) or a vectorized `memref<shapexvector<type>>` . I mean, we can extend the support for this operator if we need to.

Reply:
this bit of code looked like we are artificially limiting the support. hence the question. 
Its true that we dont exercise that but I failed to see what is special about gpu::AddressSpace::Workgroup only to support memref<shapexvector<type>>

Reply:
Yes this is limiting the support, but I don't think it would be natural to support vector global buffers . I am not even sure that `GlobalLoad` would work with `memref<shapexvector>` . And then we need to add tests for it, etc... I am happy to do so, but it seemed not useful, since we will probably never have global vector buffers